### PR TITLE
DAOS-9626 test: use crt_timeout config instead of env

### PR DESCRIPTION
**Pull request by wangshilong**
1.Python script generated crt_timeout as 30 seconds in default,
env var CRT_TIMEOUT only work if no "crt_timeout" config is
enabled, so always try to use config if we want to change
cart timeout value.

2.There is still possibility(1/10) test could teardown 
because of timeout, increase timeout.

3.Try to match long key before short key, otherwise
'>=' will match '>' firstly, it might return failure
that 'int' could not compare with str.

Signed-off-by: Wang Shilong <shilong.wang@intel.com>

### CODE REVIEW

**Review by liw — APPROVED** [review]

**Review by wiliamhuang — APPROVED** [review]

File: src/tests/ftest/rebuild/no_cap.yaml [hunk 1]
```diff
@@ -12,8 +12,7 @@ server_config:
   name: daos_server
 # reduce cart timeout to make IV update return timeout
 # quickly to ease test (DAOS-9626)
-  env_vars:
-    - CRT_TIMEOUT=10
+  crt_timeout: 10
```
Comment (liw) [inline src/tests/ftest/rebuild/no_cap.yaml:15]: Does no_cap.py need to be modified to pick up this value?

Comment (wangshilong) [reply src/tests/ftest/rebuild/no_cap.yaml:15]: No, see this example, crt_timeout is applied correctly in the daos server yml file:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8337/1/artifact/Functional%20Hardware%20Medium/rebuild/no_cap.py/1/daos_configs/daos_server.wolf-122.yml/*view*/

**Comment by wangshilong** [comment]
@daos-stack/daos-gatekeeper the only failing one looks like known issue not related to this:
** TEARDOWN called due to TIMEOUT: 210 second timeout exceeded **

File: src/tests/ftest/rebuild/no_cap.yaml (hunk 1 above)
Comment (liw) [reply src/tests/ftest/rebuild/no_cap.yaml:15]: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8337/4/artifact/Functional%20Hardware%20Medium/rebuild/no_cap.py/9/daos_logs/daos.wolf-52.log shows that the client CRT_TIMEOUT is indeed set to 10 s.

**Review by liw — APPROVED** [review]

File: src/tests/ftest/rebuild/no_cap.yaml (hunk 1 above)
Comment (wangshilong) [reply src/tests/ftest/rebuild/no_cap.yaml:15]: Client CRT_TIMEOUT seems picked from server side config, apricot/apricot/test.py:            crt_timeout=self.server_managers[index].get_config_value("crt_timeout"))

**Comment by wangshilong** [comment]
> @daos-stack/daos-gatekeeper the only failing one looks like known issue not related to this: ** TEARDOWN called due to TIMEOUT: 210 second timeout exceeded **

TEARDOWN timeout is calculated by: timeout + 30：

2022-03-05 14:48:38,679 test             L0112 INFO | self.timeout: 180
2022-03-05 14:48:38,679 test             L0120 INFO | Job-ID: job-2022-03-05T14.48-0659ef7
2022-03-05 14:48:38,679 test             L0121 INFO | Test PID: 76671
2022-03-05 14:48:38,679 test             L0497 INFO | Increasing timeout by 30 seconds for agent/server tear down: 210

There is still possibility that test could not be finished within 210 seconds, i could enlarge test timeout a bit though.

**Review by shimizukko — APPROVED** [review]

**Review by wiliamhuang — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Review by liw — APPROVED** [review]

File: src/tests/ftest/rebuild/no_cap.py [hunk 2]
```diff
@@ -154,8 +154,8 @@ def test_rebuild_no_capacity(self):
         # Check for pool and rebuild info after rebuild
         self.log.info("..(8)Verify pool and rebuild info after rebuild")
         pool_checks["pi_ndisabled"] = ">0"
-        rebuild_checks["rs_obj_nr"] = ">0"
-        rebuild_checks["rs_rec_nr"] = ">0"
+        rebuild_checks["rs_obj_nr"] = "=>0"
+        rebuild_checks["rs_rec_nr"] = "=>0"
```
Comment (phender) [inline src/tests/ftest/rebuild/no_cap.py:158]: These comparison keys for the `_check_info()` method are incorrect:
```suggestion
        rebuild_checks["rs_obj_nr"] = ">=0"
        rebuild_checks["rs_rec_nr"] = ">=0"
```

File: src/tests/ftest/util/test_utils_base.py [hunk 3]
```diff
@@ -160,18 +160,18 @@ def _check_info(self, check_list):
...
-                comparisons = {
-                    "<": (lambda x, y: x < y, "is too large"),
-                    ">": (lambda x, y: x > y, "is too small"),
-                    "<=": (
-                        lambda x, y: x <= y, "is too large or does not match"),
-                    ">=": (
-                        lambda x, y: x >= y, "is too small or does not match"),
-                }
-                for key, val in list(comparisons.items()):
+                comparisons = [
```
Comment (phender) [inline src/tests/ftest/util/test_utils_base.py:163]: Why was this file modified?  It doesn't seem to be needed as it just redefines a python dictionary as a list of lists.

**Review by phender — CHANGES_REQUESTED** [review]
Also your PR needs to run the test you are modifying.  Since the rebuild/no_cap.py test has a `daily_regression` tag it will/did not run with the default PR pragmas (see https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8337/9/testReport/FTEST_rebuild/).   To include the execution of this test you need to include the following in your commit message:
```
Features: no_cap
```

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/util/test_utils_base.py (hunk 3 above)
Comment (phender) [reply src/tests/ftest/util/test_utils_base.py:163]: Ah I see.  Thanks.

**Review by wiliamhuang — APPROVED** [review]

**Review by phender — APPROVED** [review]
